src/Segments/SegmentWithPublicName.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Napp\Xray\Segments;
+
+use Pkerrigan\Xray\Segment;
+
+class SegmentWithPublicName
+{
+    /** @var Segment */
+    public $segment;
+
+    /** @var string */
+    public $name;
```

Comment:
Use private properties with getters instead of public properties.
See also [Principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege).